src/cache/redis.ts

```diff
@@ -15,7 +15,34 @@ const logger = makeLogger('RedisCache')
 export class RedisCache<T = unknown> implements Cache<T> {
   type = CacheTypes.Redis
 
-  constructor(private client: Redis) {}
+  constructor(private client: Redis) {
+    this.loadFunctions()
+  }
+
+  async loadFunctions() {
+    // Load custom function 'setResponse' with lua script to redis that will skip overwriting successful cache response if the new value is error response
+    const source =
+      '#!lua name=ea \n ' +
+      "redis.register_function('setResponse', function(keys, args) " +
+      'local key = keys[1] ' +
+      'local value = args[1] ' +
+      'local ttl = tonumber(args[2]) ' +
+      'local json_value = cjson.decode(value) ' +
+      "local key_exists = redis.call('EXISTS', key) " +
+      'if json_value.errorMessage and key_exists == 1 then ' +
+      '  return nil ' +
+      'else ' +
+      "  return redis.call('SET', key, value, 'PX', ttl) " +
+      'end ' +
+      'end)'
+
+    const fnList = await this.client.function('LIST')
+    if (!fnList.length) {
+      this.client.function('LOAD', source)
+    } else {
+      this.client.function('LOAD', 'REPLACE', source)
+    }
```

Comment:
We cannot assume that we will be the only functions present, and the REPLACE option will override anyways, so the check is unnecessary. With the name change for the function, we can always do a function load replace